client.go

```diff
@@ -13,6 +14,7 @@ type Client struct {
 	apiKey string
 	config Config
 	client *resty.Client
+	ctx    context.Context
```

Comment:
Quote from [context godoc](https://golang.org/pkg/context/)
>  Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

For idiomatic go experience,we should add `ctx context.Context` as first argument to all public methods except `SetProxy`/`RemoveProxy`.

This will be very serious change in API, completely backward-incompatible, and context will be required, not optional as in your proposal.
On the other hand, it's a standard way of using contexts, well known to go developers.